src/query/crypt_opfunc.c

```diff
@@ -619,6 +647,67 @@ crypt_sha_functions (THREAD_ENTRY * thread_p, const char *src, int src_len, SHA_
   return NO_ERROR;
 }
 
+static int
+crypt_md5_buffer_binary (const char *buffer, size_t len, char *resblock)
+{
+  if (buffer == NULL || resblock == NULL)
+    {
+      assert (false);
+      return ER_FAILED;
+    }
+  // *INDENT-OFF*
+  deleted_unique_ptr<EVP_MD_CTX> context (EVP_MD_CTX_new (), [] (EVP_MD_CTX *ctxt_ptr)
+    {
+      if (ctxt_ptr != NULL)
+	{
+	  EVP_MD_CTX_free (ctxt_ptr); 
+	}
+    });
+  // *INDENT-ON*
+
+  if (context == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_ENCRYPTION_LIB_FAILED, 1, crypt_lib_fail_info[CRYPT_LIB_INIT_ERR]);
+      return ER_ENCRYPTION_LIB_FAILED;
+    }
+
+  if (EVP_DigestInit (context.get (), EVP_md5 ()) == 0)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_ENCRYPTION_LIB_FAILED, 1, crypt_lib_fail_info[CRYPT_LIB_INIT_ERR]);
+      return ER_ENCRYPTION_LIB_FAILED;
+    }
+  if (EVP_DigestUpdate (context.get (), buffer, len) == 0)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_ENCRYPTION_LIB_FAILED, 1, crypt_lib_fail_info[CRYPT_LIB_CRYPT_ERR]);
+      return ER_ENCRYPTION_LIB_FAILED;
+    }
+  unsigned int md_len;
+  if (EVP_DigestFinal (context.get (), (unsigned char *) resblock, &md_len) == 0)
+    {
```

Comment:
[minor] I don't think you need this because you are not using the value of the md_len variable.
`if (EVP_DigestFinal (context.get (), (unsigned char *) resblock, NULL) == 0)`

Reply:
removed md_len usage.